packages/core/src/actions/execute.ts

```diff
@@ -299,33 +300,37 @@ const findMaxBatchSize = (
     throw new Error(`Must enter at least one Merkle leaf.`)
   }
 
-  // Start with the smallest batch size (1) and incrementally try larger batches. We don't start
-  // with the largest possible batch size and incrementally try smaller batches because the gas
-  // estimation logic ABI encodes the Merkle leaves, which can be very slow for large amounts of
-  // data.
-  for (let i = 1; i <= leaves.length; i++) {
-    if (
-      !isExecutable(
-        leaves.slice(0, i),
-        maxGasLimit,
-        moduleAddress,
-        estimateGas,
-        chainId
-      )
-    ) {
-      // If the first batch itself is not executable, throw an error.
-      if (i === 1) {
-        throw new Error(`Could not find a valid batch size.`)
-      } else {
-        // For larger batches, return the size of the previous batch as the maximum executable batch
-        // size.
-        return i - 1
+  if (isActionTransactionBatchingEnabled(chainId)) {
+    // Start with the smallest batch size (1) and incrementally try larger batches. We don't start
+    // with the largest possible batch size and incrementally try smaller batches because the gas
+    // estimation logic ABI encodes the Merkle leaves, which can be very slow for large amounts of
+    // data.
+    for (let i = 1; i <= leaves.length; i++) {
+      if (
+        !isExecutable(
```

Comment:
I think we still need to call `isExecutable` for all networks, including those that have `actionTransactionBatching=false`. Otherwise, it'll be possible for us to construct a batch containing a single leaf that's greater than the max batch size limit. This is particularly a concern on Rootstock